tests/Unit/CacheInvalidatorTest.php

```diff
@@ -184,8 +184,29 @@ public function testMethodException()
      */
     public function testProxyClientExceptionsAreLogged()
     {
-        $unreachableException = ProxyUnreachableException::proxyUnreachable('http://127.0.0.1', 'Couldn\'t connect to host');
-        $responseException    = ProxyResponseException::proxyResponse('http://127.0.0.1', 403, 'Forbidden');
+        $failedRequest = \Mockery::mock('\Psr\Http\Message\RequestInterface')
+            ->shouldReceive('getHeaderLine')->with('Host')->andReturn('127.0.0.1')
+            ->getMock();
+        $adapterException = \Mockery::mock(
+            '\Http\Adapter\Exception\HttpAdapterException',
+            ['Couldn\'t connect to host']
+        )
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->getMock();
+        ;
+        $unreachableException = ProxyUnreachableException::proxyUnreachable($adapterException);
+        
+        $response = \Mockery::mock('\Psr\Http\Message\ResponseInterface')
+            ->shouldReceive('getStatusCode')->andReturn(403)
+            ->shouldReceive('getReasonPhrase')->andReturn('Forbidden')
+            ->getMock();
+        $adapterException = \Mockery::mock('\Http\Adapter\Exception\HttpAdapterException')
+            ->shouldReceive('getMessage')->andReturn('Forbidden')
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->shouldReceive('getResponse')->andReturn($response)
+            ->getMock()
+        ;
+        $responseException = ProxyResponseException::proxyResponse($adapterException);
```

Comment:
I was also thinking about how we could mock requests more easily. Maybe a library that creates requests and responses from fixtures?

For example PHP-VCR does this, but that is buried deeply in it. I have the following plan: Have a fixture generator and parser. Each objects can be created from or saved to a fixture using multiple serialization logic. For example I would like to see a message saved as it arrives (using the HTTP message format), but it could be saved do YML or JSON as well.

What do you think?